src/ui/actions/reporter.ts

```diff
@@ -40,12 +42,16 @@ export async function setupReporter(store: UIStore, ThreadFront: typeof TF) {
       )
     );
     receivedAnnotations = [];
+
+    if (finished) {
```

Comment:
Ah, this Redux store was broken before. It was marking annotations as "completed" before they were actually completed– because the handler above had been debounced:
```js
const onAnnotationsReceived = debounce(() => {
```